src/lib/actions/RunAction.ts

```diff
@@ -57,7 +58,7 @@ export class RunAction {
 			...enginePluginModules.map(pluginModule => core.dynamicallyAddEnginePlugin(pluginModule))
 		];
 		await Promise.all(addEnginePromises);
-		const workspace: Workspace = await core.createWorkspace(input.workspace);
+		const workspace: Workspace = await (new WorkspaceFactory()).create(core, input.workspace);
```

Comment:
Since `WorkspaceFactory` is just a class with implementations, we can instantiate it directly here instead of receiving one in the dependencies, which limits the cascading changes to other classes and tests. If we encounter a genuine reason that you'd want to override or stub the `WorkspaceFactory`, it's easy to convert it to the dependency pattern.

Reply:
see my other comment about this. I think we should wait to make this an official "Factory" class - since it isn't being injected into the Action classes... and because the create method is receiving yet another dependency (which is usually something you pass into the constructor of the factory). So for now, let's make these PathSTartFactory and WorkspaceFactory simple functions that we can import use and test.